### ISSUE TITLE
can not disable typecheck linter, it give annoying false report

### ISSUE THREAD

**Issue by yangjuncode** [xxpxxxxp/intellij-plugin-golangci-lint#141]
go linter 1.6.5 goland 2023.1

i have disable typecheck(it has too many false report)

.golangci.yml
```
# options for analysis running
run:
  # timeout for analysis, e.g. 30s, 5m, default is 1m
  timeout: 10m

linters:
  disable:
    - typecheck
    - unused

    # all available settings of specific linters
linters-settings:
  prealloc:
    # XXX: we don't recommend using this linter before doing performance profiling.
    # For most programs usage of prealloc will be a premature optimization.

    # Report preallocation suggestions only on simple loops that have no returns/breaks/continues/gotos in them.
    # True by default.
    simple: true
    range-loops: true # Report preallocation suggestions on range loops, true by default
    for-loops: true # Report preallocation suggestions on for loops, false by default
```

typecheck give too many false report like:
```
undefined: xsync (typecheck)

this.Lock undefined (type *TmeshControllerDevice has no field or method Lock) (typecheck)

```

go linter has recognize the conf file  .golangci.yml, but not disable typecheck linter, since typecheck give too many false report,  it's annoying.

when i run golangci-lint under cmd, it has no report.
```
[yj@yj-code01 dr086p3_server]$ golangci-lint run -v
INFO [config_reader] Config search paths: [./ /home/yj/hhh/git.a.com/a/dr086/dr086p3_server /home/yj/hhh/git.a.com/a/dr086 /home/yj/hhh/git.a.com/a /home/yj/hhh/git.a.com /home/yj/hhh /home/yj /home /] 
INFO [config_reader] Used config file .golangci.yml 
INFO [lintersdb] Active 5 linters: [errcheck gosimple govet ineffassign staticcheck] 
INFO [loader] Go packages loading at mode 575 (deps|exports_file|files|types_sizes|compiled_files|imports|name) took 245.356146ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 33.71642ms 
INFO [linters_context/goanalysis] analyzers took 0s with no stages 
INFO [runner] Issues before processing: 60, after processing: 0 
INFO [runner] Processors filtering stat (out/in): filename_unadjuster: 60/60, skip_files: 60/60, autogenerated_exclude: 14/60, cgo: 60/60, exclude-rules: 0/14, path_prettifier: 60/60, skip_dirs: 60/60, identifier_marker: 14/14, exclude: 14/14 
INFO [runner] processing took 589.078µs with stages: identifier_marker: 195.691µs, autogenerated_exclude: 166.826µs, path_prettifier: 140.065µs, skip_dirs: 52.72µs, exclude-rules: 20.909µs, cgo: 4.94µs, filename_unadjuster: 4.188µs, nolint: 551ns, max_same_issues: 491ns, fixer: 371ns, exclude: 361ns, skip_files: 290ns, severity-rules: 280ns, sort_results: 271ns, diff: 231ns, uniq_by_line: 211ns, max_from_linter: 201ns, source_code: 191ns, path_shortener: 110ns, path_prefixer: 90ns, max_per_file_from_linter: 90ns 
INFO [runner] linters took 74.431571ms with stages: goanalysis_metalinter: 73.774215ms 
INFO File cache stats: 0 entries of total size 0B 
INFO Memory: 5 samples, avg is 40.7MB, max is 54.4MB 
INFO Execution took 360.738015ms  
```

**Comment by xxpxxxxp** [comment]
Hi @yangjuncode , `typecheck` means the linter failed to parse code due to reference not found, it cannot be disabled by any means. Would you share the plugin config screenshot?

**Comment by yangjuncode** [comment]
hello @xxpxxxxp ,the screenshot are below
config
![config](https://user-images.githubusercontent.com/47293522/233763614-07951302-647f-419d-aa59-da1092bdcf00.png)
annoiying false report
![false-report](https://user-images.githubusercontent.com/47293522/233763734-faf4f92d-d7d1-40e8-997c-a89c199a03cc.png)
run fine in shell
![run-in-shell](https://user-images.githubusercontent.com/47293522/233763632-217927a5-f3bd-4ff2-a688-ab8b2488fca8.png)

the project build and run is ok.

**Comment by yangjuncode** [comment]
maybe related go mod replace, the go.mod is below
```
module dr086p3_server

go 1.19

require (
	bf8000 v0.0.0-00010101000000-000000000000
	github.com/Unknwon/goconfig v1.0.0
	github.com/bufbuild/connect-go v1.5.2
	github.com/bwmarrin/snowflake v0.3.0
	github.com/go-cmd/cmd v1.4.1
	github.com/gogo/protobuf v1.3.2
	github.com/golang/glog v1.0.0
	github.com/google/uuid v1.3.0
	github.com/gorilla/mux v1.8.0
	github.com/jackc/pgx/v4 v4.17.2
	github.com/nats-io/nats-server/v2 v2.9.12
	github.com/nats-io/nats.go v1.23.0
	github.com/puzpuzpuz/xsync/v2 v2.4.0
	github.com/soheilhy/cmux v0.1.5
	github.com/use-go/onvif v0.0.9
	github.com/uuid6/uuid6go-proto v0.2.1
	github.com/xo/dburl v0.13.0
	github.com/xtaci/kcp-go/v5 v5.6.1
	gitlab.com/zcodehelper/esrpcgateway v0.3.4
	gitlab.com/zcodehelper/zcodeutil v0.0.19
	go.uber.org/atomic v1.10.0
	google.golang.org/grpc v1.51.0
	google.golang.org/protobuf v1.30.0
	nhooyr.io/websocket v1.8.7
)

require (
	github.com/beevik/etree v1.1.0 // indirect
	github.com/elgs/gostrgen v0.0.0-20161222160715-9d61ae07eeae // indirect
	github.com/go-test/deep v1.0.8 // indirect
	github.com/gobwas/ws v1.1.0 // indirect
	github.com/gofrs/uuid v4.0.0+incompatible // indirect
	github.com/golang/protobuf v1.5.2 // indirect
	github.com/gorilla/websocket v1.4.2 // indirect
	github.com/jackc/chunkreader/v2 v2.0.1 // indirect
	github.com/jackc/pgconn v1.13.0 // indirect
	github.com/jackc/pgio v1.0.0 // indirect
	github.com/jackc/pgpassfile v1.0.0 // indirect
	github.com/jackc/pgproto3/v2 v2.3.1 // indirect
	github.com/jackc/pgservicefile v0.0.0-20200714003250-2b9c44734f2b // indirect
	github.com/jackc/pgtype v1.12.0 // indirect
	github.com/juju/errors v0.0.0-20220331221717-b38fca44723b // indirect
	github.com/klauspost/compress v1.15.15 // indirect
	github.com/klauspost/cpuid/v2 v2.0.14 // indirect
	github.com/klauspost/reedsolomon v1.10.0 // indirect
	github.com/mattn/go-isatty v0.0.16 // indirect
	github.com/minio/highwayhash v1.0.2 // indirect
	github.com/nats-io/jwt/v2 v2.3.0 // indirect
	github.com/nats-io/nkeys v0.3.0 // indirect
	github.com/nats-io/nuid v1.0.1 // indirect
	github.com/pkg/errors v0.9.1 // indirect
	github.com/rs/zerolog v1.26.1 // indirect
	github.com/smartystreets/goconvey v1.7.2 // indirect
	github.com/stretchr/testify v1.8.1 // indirect
	github.com/templexxx/cpu v0.0.9 // indirect
	github.com/templexxx/xorsimd v0.4.1 // indirect
	github.com/tjfoc/gmsm v1.4.1 // indirect
	golang.org/x/crypto v0.5.0 // indirect
	golang.org/x/exp v0.0.0-20221211140036-ad323defaf05 // indirect
	golang.org/x/net v0.5.0 // indirect
	golang.org/x/sys v0.4.0 // indirect
	golang.org/x/text v0.6.0 // indirect
	golang.org/x/time v0.3.0 // indirect
	google.golang.org/genproto v0.0.0-20221118155620-16455021b5e6 // indirect
	gopkg.in/yaml.v2 v2.4.0 // indirect
)

replace google.golang.org/grpc => github.com/yangjuncode/grpc-go v1.51.2

replace github.com/use-go/onvif => github.com/yangjuncode/onvif v0.0.10

replace github.com/xtaci/kcp-go/v5 => github.com/yangjuncode/kcp-go/v5 v5.6.2

replace bf8000 => ./bf8000

```
@xxpxxxxp 
if you want to debug, i can send you the project in private, please send me a info mail to 290968698 at qq dot com, 
i will send you the project in response.

**Comment by yangjuncode** [comment]
hey, i found the problem, i switch to windows under linux,  then it happen.

![go-setting](https://user-images.githubusercontent.com/47293522/233766916-816bfe99-6e86-42cd-9d5c-1f9e058e7136.png)
